iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1640,7 +1681,17 @@ public boolean shouldOverwrite(org.apache.hadoop.hive.ql.metadata.Table mTable,
   }
 
   @Override
-  public void addResourcesForCreateTable(Map<String, String> tblProps, HiveConf hiveConf) {
+  public void addResourcesForCreateTable(Map<String, String> tblProps, HiveConf hiveConf, Hive db,
+      TableName qualifiedTabName, boolean isExt) throws MetaException, HiveException {
+    Optional<String> defaultTableLocation =
+        SessionStateUtil.getProperty(conf, hive_metastoreConstants.DEFAULT_TABLE_LOCATION);
+    if (!defaultTableLocation.isPresent()) {
+      Warehouse wh = new Warehouse(conf);
```

Comment:
why do you need all of this? in SA#analyzeCreateTable there is already available tblLocation.
````
storageHandler.addResourcesForCreateTable(tblProps, conf, tblLocation)
````

Reply:
you can just extract in SA method `getDefaultTablePath`
````
        try {
          Warehouse wh = new Warehouse(conf);
          tblLocation = wh.getDefaultTablePath(db.getDatabase(qualifiedTabName.getDb()), qualifiedTabName.getTable(),
                  isExt).toUri().getPath();
        } catch (MetaException | HiveException e) {
          throw new SemanticException(e);
        } 
````

Reply:
Done, refactored the method & called from validateAndAddDefaultProperties